providers/jackson2/src/main/java/org/jboss/resteasy/plugins/providers/jackson/ResteasyJackson2Provider.java

```diff
@@ -256,6 +263,9 @@ public ClassLoader run()
             writer = mod.modify(endpoint, httpHeaders, value, writer, jg);
          }
 
+         // [RESTEASY-1317] Support Jackson in Atom links
+         decorateEntity(type, annotations, mediaType, new DecoratedEntityContainer(value));
```

Comment:
Can we perhaps wrap this into a if clause (maybe check the annotations and see if there's the addlink/addlinkjson one) so that we avoid creating object instances when not needed. The Jackson provider is quite commonly used, so we need to be super careful about performances / memory usage.